### PR TITLE
ci: extend the test suite

### PR DESCRIPTION
**Pull request by mrc0mmand**

### CODE REVIEW

**Comment by mrc0mmand** [comment]
As discussed on Telegram, looks like we are missing an iSCSI generator, so this test currently won't pass, because:
  * systemd doesn't start NetworkManager during boot, even though `ip=dhcp` is present, so we end up with no network - this one, imo, shouldn't depend on the iSCSI generator though?
  * the `iscsi` daemon seems to be confused as well (with `rd.systemd-wants=NetworkManager.service` to enable networking)
```
iscsi.service: Trying to enqueue job iscsi.service/reload/replace
iscsi.service: Installed new job iscsi.service/reload as 334
iscsi.service: Enqueued job iscsi.service/reload as 334
iscsi.service: Unit cannot be reloaded because it is inactive.
iscsi.service: Job 334 iscsi.service/reload finished, result=invalid
```
(see https://github.com/systemd/mkosi-initrd/runs/4957977996?check_suite_focus=true#step:5:4794)

I can boot the VM with the same command line with an initrd generated by dracut (`dracut --no-hostonly -fv -a 'iscsi qemu kernel-modules' dracut-iscsi.img --kver 5.17.0-0.rc0.20220112gitdaadb3bd0e8d.63.fc36.x86_64`).

/cc @keszybz @lnykryn

File: .github/workflows/build-fedora.sh [hunk 1]
```diff
@@ -16,44 +21,111 @@ for phase in "${PHASES[@]}"; do
...
-        INITRD_FULLBOOT)
-            if [[ ! -e "mkosi.output/initrd_$KVER.cpio.zstd" ]]; then
-                echo >&2 "Missing initrd image (run the INITRD phase first)"
-                exit 1
-            fi
+        INITRD_ISCSI)
+            rm -fr mkosi.output
+            mkdir mkosi.output
+            # Build the initrd with iSCSI support
+            python3 -m mkosi --cache "$MKOSI_CACHE" \
```
Comment (keszybz) [inline .github/workflows/build-fedora.sh]: `python3 -m` ?

File: .github/workflows/build-fedora.sh [hunk 2]
```diff
@@ -74,6 +76,56 @@ for phase in "${PHASES[@]}"; do
...
+        INITRD_LUKS)
+            rm -fr mkosi.output mkosi.extra
+            mkdir mkosi.output mkosi.extra
+
+            luks_passphrase="H3lloW0rld!"
+            # Instruct mkosi to copy the password file into the initrd image
+            # so we can use it to unlock the rootfs
+            echo -ne "$luks_passphrase" >mkosi.extra/luks.passphrase
+            # Build the initrd with dm-crypt support
+            python3 -m mkosi --cache "$MKOSI_CACHE" \
```
Comment (keszybz) [inline .github/workflows/build-fedora.sh]: Here too.

File: .github/workflows/build-fedora.sh [hunk 3]
```diff
@@ -66,6 +69,69 @@ for phase in "${PHASES[@]}"; do
...
+                                      -append "root=LABEL=root rd.debug $SYSTEMD_LOG_OPTS console=ttyS0 systemd.unit=systemd-poweroff.service systemd.default_timeout_start_sec=240"
+
+            # Cleanup
+            rm -fr mkosi.output _rootfs
+            ;;
+        INITRD_LVM)
+            rm -fr mkosi.output
+            mkdir mkosi.output
+            # Build the initrd with LVM support
+            python3 -m mkosi --cache "$MKOSI_CACHE" \
```
Comment (keszybz) [inline .github/workflows/build-fedora.sh]: Ditto.

File: .github/workflows/build-fedora.sh [hunk 4]
```diff
@@ -66,6 +69,69 @@ for phase in "${PHASES[@]}"; do
...
+            # shellcheck source=/dev/null
+            source <(grep -E "(ID|VERSION_ID)" /etc/os-release)
+            mkosi --cache "$MKOSI_CACHE" \
+                  --distribution="$ID" \
+                  --release="$VERSION_ID" \
+                  --format=directory \
+                  --output=out
+            # Note: this is necessary, since mkosi requires the target directory
+            #       to not exist and we don't want it to remove the mnt mountpoint
+            (shopt -s dotglob && mv out/* mnt/)
```
Comment (keszybz) [inline .github/workflows/build-fedora.sh:141]: RFE for `mkosi` to allow this?

File: .github/workflows/build-fedora.sh [hunk 5]
```diff
@@ -192,6 +192,80 @@ for phase in "${PHASES[@]}"; do
...
+        INITRD_LUKS_LVM)
+            rm -fr mkosi.output mkosi.extra
+            mkdir mkosi.output mkosi.extra
+
+            luks_passphrase="H3lloW0rld!"
+            # Instruct mkosi to copy the password file into the initrd image
+            # so we can use it to unlock the rootfs
+            echo -ne "$luks_passphrase" >mkosi.extra/luks.passphrase
+            # Build the initrd with LVM support
+            python3 -m mkosi --cache "$MKOSI_CACHE" \
```
Comment (keszybz) [inline .github/workflows/build-fedora.sh]: Ditto.

**Review by keszybz — COMMENTED** [review]
This is great!

(Looking at all the code required for the tests, I think that we might want to extract some parts of it to be standalone to allow running locally, and then just call that from the github workflows. But this something for later. Let's get things going via github actions first.)

File: .github/workflows/build-fedora.sh (hunk 1 above)
Comment (mrc0mmand) [reply .github/workflows/build-fedora.sh]: Hah, force of habit I guess from local testing, replaced in the new tests, but also in the existing ones.

File: .github/workflows/build-fedora.sh (hunk 4 above)
Comment (mrc0mmand) [reply .github/workflows/build-fedora.sh:141]: Filed as https://github.com/systemd/mkosi/issues/1251.

**Pull request merged**